x/gov/keeper_test.go

```diff
@@ -0,0 +1,205 @@
+package gov
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func TestGetSetProposal(t *testing.T) {
+
+	ctx, _, keeper := createTestInput(t, false, 100)
+
+	proposal := keeper.NewProposal(ctx, "Test", "description", "Text")
+	proposalID := proposal.ProposalID
+	keeper.SetProposal(ctx, proposal)
+
+	gotProposal := keeper.GetProposal(ctx, proposalID)
+
+	assert.Equal(t, proposal.ProposalID, gotProposal.ProposalID)
+	assert.Equal(t, proposal.Title, gotProposal.Title)
+	assert.Equal(t, proposal.Description, gotProposal.Description)
+	assert.Equal(t, proposal.ProposalType, gotProposal.ProposalType)
+	assert.Equal(t, proposal.SubmitBlock, gotProposal.SubmitBlock)
+	assert.True(t, proposal.TotalDeposit.IsEqual(gotProposal.TotalDeposit))
+	assert.Equal(t, proposal.SubmitBlock, gotProposal.SubmitBlock)
```

Comment:
should create a `proposal.Equal` function to consolidate the above statements into one